presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/MergeLimitWithDistinct.java

```diff
@@ -35,7 +35,18 @@
     private static final Capture<AggregationNode> CHILD = newCapture();
 
     private static final Pattern<LimitNode> PATTERN = limit()
-            .with(source().matching(aggregation().capturedAs(CHILD)));
+            .with(source().matching(aggregation().capturedAs(CHILD)
+                    .matching(aggregation -> isDistinct(aggregation))));
```

Comment:
I tried and changed to this, as this is more readable than `MergeLimitWithDistinct::isDistinct`